fvtest/compilerunittest/aarch64/BinaryEncoder.cpp

```diff
@@ -101,6 +102,25 @@ TEST_P(ARM64VectorShiftImmediateEncodingTest, encode) {
     ASSERT_EQ(std::get<4>(GetParam()), encodeInstruction(instr));
 }
 
+class ARM64VectorSplatsImmediateEncodingTest : public TRTest::BinaryEncoderTest<ARM64_INSTRUCTION_ALIGNMENT>, public ::testing::WithParamInterface<std::tuple<TR::InstOpCode::Mnemonic, TR::RealRegister::RegNum, TR::DataType, uint64_t, ARM64BinaryInstruction>> {};
+
+TEST_P(ARM64VectorSplatsImmediateEncodingTest, encode) {
+    auto trgReg = cg()->machine()->getRealRegister(std::get<1>(GetParam()));
+    auto op = std::get<0>(GetParam());
+    auto elementType = std::get<2>(GetParam());
+    auto childNode = TR::Node::create(TR::iconst);
+    childNode->setConstValue(std::get<3>(GetParam()));
+
+    auto instr = TR::TreeEvaluator::vsplatsImmediateHelper(fakeNode, cg(), childNode, elementType, trgReg);
+
+    if (op == TR::InstOpCode::bad) {
```

Comment:
What is your intention for adding tests for `TR::InstOpCode::bad`?

Reply:
The intention is to check that `vsplatsImmediateHelper` returns NULL for the constant value which cannot be encoded into movi or mvni instruction. `TR::InstOpCode::bad` means we cannot encode the specified constant value.